### PR TITLE
[4.0] add aria-label / aria-labelledby to module rendering

### PR DESCRIPTION
**Pull request by hans2103**
Pull Request for Issue # .

### Summary of Changes

Make Joomla more accessible by adding the attributes `aria-label` or `aria-labelledby`
The `aria-label` attribute is used to define a string that labels the current element. Use it in cases where a text label is _not_ visible on the screen. If there is visible text labeling the element, use `aria-labelledby` instead.

This attribute can be used with any typical HTML element; it is not limited to elements that have an ARIA `role` assigned.

source: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-label_attribute

When adding a module people can choose the module element. This can be a `div` element, but a `section` or an `aside` element too. The latter are called Landmarks. When adding more landmarks of the same type you have to add a label. Otherwise screenreaders have no clue about them. 

Since Cassiopeia will be used to inspire people I have chosen to use the `ArrayHelper::toString` to render the attributes of the element (moduleTag or headerTag) stored in moduleAttribs and headerAttribs.

This PR will not change the current look and feel of the Cassiopeia template.

### Testing Instructions

This PR only has influence on the module type html5, Cassiopeia card and Cassiopeia cardGrey.
Apply the PR and view source code of the homepage. 
- attribute `aria-label` is applied to modules where title is set to hide
- attribute `aria-labelledby` is applied to module where title is set to show

### Actual result BEFORE applying this Pull Request

No attribute `aria-label` or `aria-labelledby` are present in the modules of type html5, Cassiopeia card nor Cassiopeia cardGrey.

### Expected result AFTER applying this Pull Request

The attribute `aria-label` or `aria-labelledby` are present in the modules of type html5, Cassiopeia card nor Cassiopeia cardGrey.
- The attribute `aria-label` will be used when module parameter `title` is set to hide. 
- The attribute `aria-labelledby` will be used when module parameter `title` is set to show. The value refers to the newly set attribute `id` in the displayed title

### Documentation Changes Required

No changes required

### CODE REVIEW

**Comment by brianteeman** [comment]
This fails validation because it should be `aria-labelledby` 

Two letter L not one

**Comment by brianteeman** [comment]
Otherwise it works as intended - nice stuff

**Comment by hans2103** [comment]
> This fails validation because it should be `aria-labelledby`
> 
> Two letter L not one

thank you @brianteeman
typo fixed and PR updated

**Comment by hans2103** [comment]
PR updated with latest changes from branch upstream/4.0dev

**Comment by ceford** [comment]
I have tested this item :white_check_mark: successfully on f2276b6e39738d12efdba647a9c4ae331facc8cc<br /><br />Without the patch there was one instance of aria-label
		<nav role="navigation" aria-label="Breadcrumbs">
With the patch there were 14 instances of aria-labelledby
				<div class="moduletable " aria-labelledby="mod-138">
And 5 instances of aria-label
		<div class="top-a card card-grey " aria-label="Main Menu Blog (en-GB)">
(A lot of experimentation with modules)<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/29886">issues.joomla.org/tracker/joomla-cms/29886</a>.</sub>

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on f2276b6e39738d12efdba647a9c4ae331facc8cc<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/29886">issues.joomla.org/tracker/joomla-cms/29886</a>.</sub>

**Comment by alikon** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/29886">issues.joomla.org/tracker/joomla-cms/29886</a>.</sub>

File: layouts/chromes/html5.php [hunk 1]
```diff
@@ -9,6 +9,8 @@
  * html5 (chosen html5 tag and font header tags)
  */
 
+use Joomla\Utilities\ArrayHelper;
```
Comment (Quy) [inline layouts/chromes/html5.php]: Swap lines 12 and 14.

File: layouts/chromes/html5.php [hunk 2]
```diff
@@ -19,19 +21,28 @@
...
 
-$moduleTag     = htmlspecialchars($params->get('module_tag', 'div'), ENT_QUOTES, 'UTF-8');
-$headerTag     = htmlspecialchars($params->get('header_tag', 'h3'), ENT_QUOTES, 'UTF-8');
-$bootstrapSize = (int) $params->get('bootstrap_size', 0);
-$moduleClass   = $bootstrapSize !== 0 ? ' col-md-' . $bootstrapSize : '';
+$moduleTag              = htmlspecialchars($params->get('module_tag', 'div'), ENT_QUOTES, 'UTF-8');
+$moduleAttribs          = [];
+$moduleAttribs['class'] = 'moduletable ' . htmlspecialchars($params->get('moduleclass_sfx'), ENT_QUOTES, 'UTF-8');
+$bootstrapSize          = (int) $params->get('bootstrap_size', 0);
+$moduleAttribs['class'] .= $bootstrapSize !== 0 ? ' col-md-' . $bootstrapSize : '';
```
Comment (Quy) [inline layouts/chromes/html5.php:28]: Why concatenate on array?

Comment (hans2103) [reply layouts/chromes/html5.php:28]: @Quy lack of decent PHP knowledge I guess. Any suggestions to improve it?

File: layouts/chromes/html5.php (hunk 1 above)
Comment (hans2103) [reply layouts/chromes/html5.php]: solved in commit 1aba84e261a0f150f19818a51b02077a70a6893a

Comment (Quy) [reply layouts/chromes/html5.php]: Please repeat for the other instances in this PR.

File: layouts/chromes/html5.php (hunk 2 above)
Comment (Quy) [reply layouts/chromes/html5.php:28]: Remove the period, but I will have to review the PR later to see what is going on.

Comment (hans2103) [reply layouts/chromes/html5.php:28]: @Quy If I remove the period all previous set content of `$moduleAttribs['class']` will be lost. That is not what I want. I want to add the bootstrap_size, if set.

Comment (Quy) [reply layouts/chromes/html5.php:28]: Lines 26 & 28 have the same array slot so line 28 overwrites line 26.

File: templates/cassiopeia/html/layouts/chromes/cardGrey.php [hunk 3]
```diff
@@ -7,6 +7,8 @@
  * @license     GNU General Public License version 2 or later; see LICENSE.txt
  */
 
+use Joomla\Utilities\ArrayHelper;
+
 defined('_JEXEC') or die;
```
Comment (Quy) [inline templates/cassiopeia/html/layouts/chromes/cardGrey.php]:
```suggestion
defined('_JEXEC') or die;

use Joomla\Utilities\ArrayHelper;
```

File: templates/cassiopeia/html/layouts/chromes/default.php [hunk 4]
```diff
@@ -7,6 +7,8 @@
  * @license     GNU General Public License version 2 or later; see LICENSE.txt
  */
 
+use Joomla\Utilities\ArrayHelper;
+
 defined('_JEXEC') or die;
```
Comment (Quy) [inline templates/cassiopeia/html/layouts/chromes/default.php]:
```suggestion
defined('_JEXEC') or die;

use Joomla\Utilities\ArrayHelper;
```

File: layouts/chromes/html5.php (hunk 2 above)
Comment (Quy) [reply layouts/chromes/html5.php:28]: Nevermind. Leave it as it.

Comment (brianteeman) [reply layouts/chromes/html5.php:28]: .= means that it is appended doesnt it? so nothing is overwritten at all

File: templates/cassiopeia/html/layouts/chromes/cardGrey.php (hunk 3 above)
Comment (hans2103) [reply templates/cassiopeia/html/layouts/chromes/cardGrey.php]: fixed in 3aae9c4cdbe0a534f6db485b8168ef3aa6103138

File: templates/cassiopeia/html/layouts/chromes/default.php (hunk 4 above)
Comment (hans2103) [reply templates/cassiopeia/html/layouts/chromes/default.php]: fixed in 3aae9c4cdbe0a534f6db485b8168ef3aa6103138

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**